### PR TITLE
Restructured machine learning section

### PR DESCRIPTION
**Pull request by cyenyxe**
Extended introduction by combining with content previously on the ledger section.

First tutorial, about using the machine library in a standalone application, moved to the quickstart.

### CODE REVIEW

File: docs/getting-started/quickstart.md [hunk 1]
```diff
@@ -113,9 +113,83 @@ This is intentionally very simple; to see a full example go <a href="https://git
...
+    var test_label = readCSV(System.Argv(4));
+
+endfunction
+```
+
+### Setting up the model
+
+We then create a `Model` type and set a flag on it that describes what kind of `Model` we want. A `sequential` model allows us to manually add layers specifying:
+
+* Type: `dense` in this example
```
Comment (baykaner) [inline docs/getting-started/quickstart.md]:
```suggestion
* Type: `dense` in this example refers to a fully-connected layer where every neuron in the layer is connected to every input and every output.
```

File: docs/getting-started/quickstart.md [hunk 2]
```diff
@@ -113,9 +113,83 @@ This is intentionally very simple; to see a full example go <a href="https://git
...
+* Input size: must match the data for the first layer
+* Neurons in layer / output size: must match the label size for the final layer
+* Activation type (optional): here we use a rectified linear unit for the first two layers, and no activation for the final layer
+
+Finally, we compile the model using the `mean squared error` loss function, and the `adam` optimiser.
```
Comment (baykaner) [inline docs/getting-started/quickstart.md]:
```suggestion
Finally, we compile the model using the `mse` (mean squared error) loss function, and the `adam` optimiser.
```

File: docs/machine-learning/index.md [hunk 3]
```diff
@@ -1,15 +1,11 @@
 # Machine Learning
 
-The Fetch.ai distributed ledger is fully compatible with 
-Fetch's machine learning library. That means neural networks can be trained and run on or off chain 
-depending on what is needed for your application.
+Fetch.ai has developed a machine learning library is a generalised toolkit for building and running this type of applications on distributed ledger nodes. This means neural networks can be trained and run on or off chain depending on what is needed for your application. Machine learning operations are implemented in the Etch language, used to write smart contracts native to the Fetch.ai platform.
```
Comment (baykaner) [inline docs/machine-learning/index.md]:
```suggestion
Fetch.ai has developed a machine learning library that is a generalised toolkit for building and running these type of applications on distributed ledger nodes. This means neural networks can be trained and run on or off chain depending on what is needed for your application. Machine learning operations may be invoked with the Etch language, used to write smart contracts native to the Fetch.ai platform.
```

File: docs/machine-learning/index.md [hunk 4]
```diff
@@ -1,15 +1,11 @@
...
 
-If you're new to the Etch language, used to write smart contracts native to the Fetch.ai platform,
-<a href="/machine-learning/basic-etch-example" target=_blank>this etch example</a> demonstrates how to train and run a neural network.
-This can be useful for trying out ideas, or simply using the fetch machine learning library for standalone applications.
+Like PyTorch and TensorFlow, the library is focused on neural networks and deep learning. But in contrast with them, it is compatible with Etch `FixedPoint` data types and Tensors built with them. These data types implement integer mathematics, wich make possible to compute precise results of complex mathematical operations (in line with deep neural networks requirements) on different architectures with guaranteed identical results.
```
Comment (baykaner) [inline docs/machine-learning/index.md]:
```suggestion
Like PyTorch and TensorFlow the library focuses on building, training, and running computational graphs, in order to implement neural networks for deep learning. In contrast with these frameworks, the Fetch machine learning library is fully compatible with integer mathematics, implemented with Etch `FixedPoint` data types (and Tensors containing them). As a result every neural network (and every complex mathematical operation in the library) will produce numerically identical results across the full range of hardware and software architectures executing the library. This means that every machine learning operation taking place on the ledger will verifiable by every other node in the network.
```

File: docs/machine-learning/index.md [hunk 5]
```diff
@@ -1,15 +1,11 @@
...
 
-To extending this further we need to understand how to train or run a neural net on-chain. 
-<a href="/machine-learning/smart-contract-example" target=_blank>This smart contract example </a> demonstrates how to rework this neural network to train and run on chain.
+If you are new to Etch, the language used to write smart contracts native to the Fetch.ai platform, [the quickstart section](/getting-started/quickstart#training-a-neural-network) demonstrates how to train and run a neural network. This can be useful for trying out ideas, or simply using the Fetch.ai machine learning library for standalone applications.
 
-Finally, <a href="/machine-learning/synergetic-contract-example" target=_blank>this synergetic contract example</a> covers synergetic contracts; this is useful for coordinating between on-chain and off-chain work.
-This is great for executing expensive computations off-chain but coordinating/managing the work on-chain. 
+To extending this further we need to understand how to train or run a neural net on-chain. [This smart contract example](/machine-learning/smart-contract-example) demonstrates how to rework this neural network to train and run on chain.
```
Comment (baykaner) [inline docs/machine-learning/index.md]:
```suggestion
To extend this further we need to understand how to train or run a neural net on-chain. [This smart contract example](/machine-learning/smart-contract-example) demonstrates how to rework this neural network to train and run on chain.
```

**Review by baykaner — CHANGES_REQUESTED** [review]

File: docs/machine-learning/index.md (hunk 4 above)
Comment (cyenyxe) [reply docs/machine-learning/index.md]:
```suggestion
Like PyTorch and TensorFlow the library focuses on building, training, and running computational graphs, in order to implement neural networks for deep learning. In contrast with these frameworks, the Fetch.ai machine learning library is fully compatible with integer mathematics, implemented with Etch `FixedPoint` data types (and Tensors containing them). As a result, every neural network and complex mathematical operation in the library will produce numerically identical results across any hardware and software architecture. This means that every machine learning operation taking place on the ledger will be verifiable by every other node in the network.
```
